src/component/category-form/category-form.js

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import autoBind from '../../utils/';
+
+// this is the UI state. Everything but the store is the UI state from now on.
+const defaultState = {
+  title: '',
+};
+
+class CategoryForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = this.props.category || defaultState;
+    autoBind.call(this, CategoryForm);
+  }
+
+  handleChange(event) {
+    const { name, value } = event.target;
+    this.setState({ [name]: value });
```

Comment:
`[name]` references your attribute `name=` in the input element, not the `'name'` value given to the attribute, this is why you are getting the console error. If you want more clarification on this, just ask :)